MapboxNavigation/NavigationMapView.swift

```diff
@@ -1335,3 +1367,122 @@ open class NavigationMapView: MGLMapView, UIGestureRecognizerDelegate {
+    
+    static func ==(lhs: BuildingHighlightAttributes, rhs: BuildingHighlightAttributes) -> Bool {
+        return lhs.identifier == rhs.identifier
+    }
+}
+
+extension NavigationMapView {
+       
+    /**
+     Receives coordinates for searching the map for buildings. If buildings are found, they will be highlighted in 2d or 3d depending on the `in3D` value.
+     `in3D` defaults to true
+     `buildingHighlightingEnabled` must be set to true before this function will work.
+     */
+    public func highlightBuildings(at coordinates: [CLLocationCoordinate2D], in3D extrudesBuildings: Bool = true) {
+        let buildingAttributes = coordinates
+            // Create building attributes by searching for building ID by provided coordinate.
+            .map({ BuildingHighlightAttributes(coordinate: $0,
+                                               highlightColor: buildingHighlightColor,
+                                               identifier: buildingId(at: $0)) })
+            // Do not add building attribute, in case if it wasn't found.
+            .filter({ $0.identifier != BuildingHighlightAttributes.kDefaultIdentifier })
+            // Since there might be case in which the same building is highlighted several times, remove duplicated IDs.
+            .withoutDuplicates
+        
+        highlightBuildings(buildingAttributes, in3D: extrudesBuildings, extrudeAll: false)
+    }
+    
+    /**
+     Removes the highlight from all buildings highlighted by `highlightBuildings(for coordinates:, in3D:)`.
+     */
+    public func unhighlightBuildings() {
+        guard let highlightedBuildingsLayer = style?.layer(withIdentifier: StyleLayerIdentifier.buildingExtrusion) else { return }
+        
+        style?.removeLayer(highlightedBuildingsLayer)
+    }
+    
+    private func addBuildingsLayer() -> MGLFillExtrusionStyleLayer? {
+        if let highlightedBuildingsLayer = style?.layer(withIdentifier: StyleLayerIdentifier.buildingExtrusion) as? MGLFillExtrusionStyleLayer { return highlightedBuildingsLayer }
+        guard let buildingsSource = style?.source(withIdentifier: "composite") else { return nil }
+        
+        let highlightedBuildingsLayer = MGLFillExtrusionStyleLayer(identifier: StyleLayerIdentifier.buildingExtrusion, source: buildingsSource)
+        highlightedBuildingsLayer.sourceLayerIdentifier = "building"
+        highlightedBuildingsLayer.fillExtrusionColor = NSExpression(forConstantValue: buildingDefaultColor)
+        highlightedBuildingsLayer.fillExtrusionOpacity = NSExpression(forConstantValue: 0.05)
+        highlightedBuildingsLayer.fillExtrusionHeightTransition = MGLTransition(duration: 0.8, delay: 0)
+        highlightedBuildingsLayer.fillExtrusionOpacityTransition = MGLTransition(duration: 0.8, delay: 0)
+        
+        style?.addLayer(highlightedBuildingsLayer)
+        
+        return highlightedBuildingsLayer
+    }
+
+    private func buildingId(at coordinate: CLLocationCoordinate2D) -> Int64 {
```

Comment:
Nit: Spell out “identifier” instead of [abbreviating](https://swift.org/documentation/api-design-guidelines/#avoid-abbreviations) it as “id”. That will also skirt any inconsistency between “ID” and “Id”.